### ISSUE TITLE
Double quotes being removed from headers with single quotes, leading to rejections

### ISSUE THREAD

**Issue by WesCossick** [nodemailer/nodemailer#1632]
Within the last couple of weeks, Yahoo has begun rejecting emails whose `From` header contains an unquoted single quote. For example, if the `From` header looks like:

```
From: Example's Thing <user@example.com>
```

Then the message will be rejected with the following SMTP error code:

```
SMTP; 554 5.7.9 Message not accepted for policy reasons. See https://postmaster.yahooinc.com/error-codes
```

_(Weirdly, this is only happening for the non-`yahoo.com` domains that Yahoo manages, like `att.net` and `bellsouth.net`.)_

Our engineering team has been in communication with Yahoo's engineering team, and we've confirmed that the issue is that Yahoo has just started requiring the sender's name to be wrapped in double quotes if it contains a single quote.

Here's the problem, though. If we pass the following `from` option to Nodemailer...

```
from: `"Example's Thing" <user@example.com>`
```

...Nodemailer strips the double quotes and turns the header into...

```
From: Example's Thing <user@example.com>
```

...which then gets rejected by Yahoo. Nodemailer doesn't always strip the double quotes, though. If the `from` option contains escaped double quotes, Nodmailer seems to retain the double quotes around the sender's name.

With the new policy from Yahoo, we need Nodemailer to retain the double quotes if the sender's name contains a single quote.

**Comment by WesCossick** [comment]
Although we haven't specifically tested or confirmed this, I assume Nodemailer is also removing double quotes around the recipient's name if it contains a single quote, and I suspect this would also lead to a similar rejection by Yahoo.

**Comment by andris9** [comment]
That's good to know. Nodemailer does not remove quotes, but it parses addresses into normalized values and then recombines these into the final address string and only uses double quotes when needed. I published a new version (v6.9.12) that treats the single quote as a special character that requires quoting.

**Comment by WesCossick** [comment]
Thank you!
